core/modules/qdisp/QueryRequest.cc

```diff
@@ -71,29 +70,48 @@ QueryRequest::~QueryRequest() {
 
 // content of request data
 char* QueryRequest::GetRequest(int& requestLength) {
-    requestLength = _jobDesc.payload().size();
-    LOGF_DEBUG("Requesting, payload size: [%1%]" % requestLength);
+    std::lock_guard<std::mutex> lock(_finishStatusMutex);
+    if (_finishStatus != ACTIVE) {
+        LOGF_WARN("%1% QueryRequest::GetRequest called after job finished (cancelled?)" % _jobId);
+        return const_cast<char*>("");
```

Comment:
Yuck. Of course we should blame xrootd for this but maybe we can avoid cast, e.g.:

    static char* empty = "";
    return empty;